src/box/replication.cc

```diff
@@ -288,7 +288,7 @@ replica_check_id(uint32_t replica_id)
 static bool
 replica_is_orphan(struct replica *replica)
 {
-	return replica->id == REPLICA_ID_NIL &&
+	return replica->id == REPLICA_ID_NIL && replica->gc == NULL &&
```

Comment:
AFAIU any anonymous replica has a garbage collector state now so shouldn't it be `&& !replica->anon`? Anyway, the comment needs to be updated.

Reply:
> Anyway, the comment needs to be updated.

Thanks, done.

> AFAIU any anonymous replica has a garbage collector state now so shouldn't it be && !replica->anon?

Why? Then anonymous replica will always be non-orphan and won't be deleted at all.

Reply:
Anonymous replicas should be deleted only on timeout, i.e. they should never be considered orphan, no?

Reply:
Fiber `replication_anon_gc` calls general-purpose `replica_deactivate` function that deletes the replica only if it is orphan (since this function can be called manually for regular replicas). We could do something like
```c
if (replica_is_orphan(replica) || replica->anon) {
    ...
}
```
but it looks weird, we always use delete-replica-if-orphan construction without additional conditions.

Reply:
How can an anonymous replica become orphan? Its gc state should be deleted but it can only happen in `replication_anon_gc` or do I miss somehing?

Reply:
Discussed f2f, got your point - now all anonymous replicas are not orphan.